kobo/apps/subsequences/tests/test_nlp_integration.py

```diff
@@ -0,0 +1,130 @@
+import pytest
+
+from django.conf import settings
+from django.test import TestCase
+from django.utils import timezone
+from model_bakery import baker
+from unittest.mock import patch, Mock
+
+from kpi.models import Asset
+from ..actions.base import ACTION_NEEDED, PASSES
+from ..actions.automatic_transcription import (
+    AutomaticTranscriptionAction,
+    REQUESTED_BY_USER,
+    PENDING,
+)
+from ..constants import GOOGLETS, GOOGLETX
+from ..models import SubmissionExtras
+
+TEST_TRANSCRIPTION_SERVICES = [
+    'acme_1_speech2text',
+    'optimus_transcribers',
+    'wonka_stenographers',
+]
+
+
+class NLPIntegrationTestCase(TestCase):
+
+    fixtures = ['test_data']
+
+    def setUp(self):
+        user = baker.make(
+            settings.AUTH_USER_MODEL,
+            username='johndoe',
+            date_joined=timezone.now().today,
+        )
+
+        self.asset = Asset.objects.create(
+            owner=user, content={
+                'survey': [
+                    {'type': 'audio', 'name': 'ask_a_question'}
+                ]
+            }
+        )
+        self.asset.advanced_features = {
+            'transcript': {'languages': ['en']},
+            'translation': {'languages': ['en', 'es']},
+        }
+
+    def test_param_builder(self):
+        AutomaticTranscriptionAction.TRANSCRIPTION_SERVICES = TEST_TRANSCRIPTION_SERVICES
+        survey = self.asset.content
+        built_params = AutomaticTranscriptionAction.build_params({}, survey)
+        assert built_params['values'] == ['ask_a_question']
+        assert 'services' in built_params
+
+
+    def test_instantiate_action_with_params(self):
+        survey = self.asset.content
+        action_params = AutomaticTranscriptionAction.build_params({}, survey)
+        action_instance = AutomaticTranscriptionAction(action_params)
+        assert action_instance is not None
+
+    @pytest.mark.skip(reason='transcription currently does not depend on this working')
```

Comment:
Is it something that comes from former PR? 
If not, I don't understand the reason why this test is skipped?

Reply:
Yes, this is testing a feature that is not really implemented. The Actions architecture that was made for the subsequences app is supposed to handle pending actions through Actions classes. So what I understand is this: instead of handling the processing on the save override for the submission extras model, they could be handled automatically through these actions architecture. However, the translation and transcription actions are not really implemented that way. And the save override doesn't use the actions interface to execute any pending actions. Right now the actions classes are being used only to specify the JSON schema modifications done to the submission extra data. 

So I think I could adapt the current solution to this actions architecture in order to avoid having this tech debt hanging in there.

Reply:
sounds a good idea ;-) 
@jnm  what do you think?

Reply:
We had a chat about this and decided to leave it alone for now. We will have to pay off the tech debt eventually, but it's a higher priority to address other NLP-related tech debt like https://www.notion.so/kobotoolbox/Pick-different-delimiter-for-qpath-probably-with-URL-encoding-e1271ab8ec4442bcb7c5af231f751127

Reply:
Ok, thanks John and Olivier for the feedback.